vim-patch:8.2.{2198,3159}

janlazo opened the pull request:
Problem:    ml_get error when resizing window and using text property.
Solution:   Validate botline of the right window. (closes vim/vim#7528)

https://github.com/vim/vim/commit/23999d799cfe844b604f193183f8f84052c8e746

<!--
  Thank you for contributing to Neovim!
  If this is your first time, check out https://github.com/neovim/neovim/blob/master/CONTRIBUTING.md#pull-requests-prs
  for our PR guidelines.
-->

janlazo committed:
vim-patch:8.2.2198: ml_get error when resizing window and using text property

Problem:    ml_get error when resizing window and using text property.
Solution:   Validate botline of the right window. (closes vim/vim#7528)

https://github.com/vim/vim/commit/23999d799cfe844b604f193183f8f84052c8e746

Migrate to Vim's (in)validate_botline_win() API.
Nvim wants to pass "curwin" instead of hiding them
behind alias/macro/inline-function.

https://github.com/neovim/neovim/pull/37164#discussion_r2655006908

Co-authored-by: Bram Moolenaar <Bram@vim.org>

zeertzjq reviewed (commented):
This makes `validate_botline()` inconsistent with `invalidate_botline()`. `invalidate_botline_win()` was renamed to `invalidate_botline()` in 008154954791001efcc46c28146e21403f3a698b.

janlazo committed:
vim-patch:8.2.3159: cursor displayed in wrong position after deleting line

Problem:    Cursor displayed in wrong position after deleting line.
Solution:   When deleting lines do not approximate botline. (fixes vim/vim#8559)

https://github.com/vim/vim/commit/5bea41dea34bcc73c4efb1b554ad0a2018966ecc

Co-authored-by: Bram Moolenaar <Bram@vim.org>

bfredl commented on src/nvim/move.c:
```diff
@@ -608,16 +611,30 @@ void changed_line_abv_curs_win(win_T *wp)
+void validate_botline(void)
+{
+  validate_botline_win(curwin);
+}
```
We don't want these alias functions back. caller should spell out `curwin` if that is what is intended.

zeertzjq commented:
I think these should be `refactor(move.c)`, not `fix(move.c)`
